app/controllers/Tournament.scala

```diff
@@ -162,14 +162,14 @@ object Tournament extends LilaController {
       NoPlayban {
         val password = ctx.body.body.\("p").asOpt[String]
         negotiate(
-          html = repo enterableById id map {
-            case None => tournamentNotFound
+          html = repo enterableById id flatMap {
+            case None => fuccess(tournamentNotFound)
             case Some(tour) =>
-              env.api.join(tour.id, me, password)
-              Redirect(routes.Tournament.show(tour.id))
+              env.api.join(tour.id, me, password, getUserTeamIds)
+              fuccess(Redirect(routes.Tournament.show(tour.id)))
```

Comment:
looks like the `flatMap` and `fuccess`s are not necessary, you can just keep the `map`.